lib/dest/writeContents/index.js

```diff
@@ -1,11 +1,23 @@
 'use strict';
 
-var fs = require('fs');
+var fs = require('graceful-fs');
+var assign = require('object-assign');
+
 var writeDir = require('./writeDir');
 var writeStream = require('./writeStream');
 var writeBuffer = require('./writeBuffer');
 var writeSymbolicLink = require('./writeSymbolicLink');
 
+// TODO include sticky/setuid/setgid, i.e. 7777?
+var MASK_MODE = parseInt('0777', 8);
```

Comment:
@erikkemperman  what would happen if this was just `'777'` like we did at https://github.com/gulpjs/vinyl-fs/blob/master/test/dest.js#L38